test/com/google/javascript/jscomp/CoverageInstrumentationPassTest.java

```diff
@@ -165,4 +165,31 @@ public void testEsModule() throws Exception {
             LanguageMode.ECMASCRIPT_NEXT, LanguageMode.ECMASCRIPT_NEXT, /* coverageOnly= */ true),
         "CoverageInstrumentationPassTest/EsModule.jsdata");
   }
+
+  @Test
+  public void testNestedForLoop() throws Exception {
+    GoldenFileComparer.compileAndCompare(
+        "CoverageInstrumentationPassTest/NestedForLoopBranchGolden.jsdata",
+        branchOptions(LanguageMode.STABLE),
+        "CoverageInstrumentationPassTest/NestedForLoopBranch.jsdata");
+  }
+
+  @Test
+  public void testForLoopNoEndingBlock() throws Exception {
+    GoldenFileComparer.compileAndCompare(
+        "CoverageInstrumentationPassTest/ForLoopBranchNoEndingBlockGolden.jsdata",
+        branchOptions(LanguageMode.STABLE),
+        "CoverageInstrumentationPassTest/ForLoopBranchNoEndingBlock.jsdata");
+  }
+
+  @Test
+  public void testForLoopConflictWithPolyfill() throws Exception {
+    CompilerOptions options = branchOptions(LanguageMode.STABLE);
+    options.setRewritePolyfills(true);
```

Comment:
Consider using `options.forceLibraryInjection` instead of `options.RewritePolyfills` to make this less brittle.

Also I'd suggest adding a comment on what exactly this case covers and explaining that it will become less useful if the relevant polyfill file changes.

Reply:
Updated.